From 0e8ea14cca825e30e5b4b21afd012e39aedb35c9 Mon Sep 17 00:00:00 2001
From: naz664 <nazimnavas145@gmail.com>
Date: Sat, 28 Oct 2023 13:51:59 +0000
Subject: [PATCH 1/2] Revert "BrightnessUtils: Conditionally use low gamma
 implementation for brightness curve"

This reverts commit 3840e36dc962aabbf4c873793d887ce7ce245edc.
---
 .../settingslib/display/BrightnessUtils.java     | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/display/BrightnessUtils.java b/packages/SettingsLib/src/com/android/settingslib/display/BrightnessUtils.java
index bbfd8cefcddf..4f86afaa995c 100644
--- a/packages/SettingsLib/src/com/android/settingslib/display/BrightnessUtils.java
+++ b/packages/SettingsLib/src/com/android/settingslib/display/BrightnessUtils.java
@@ -16,18 +16,12 @@
 
 package com.android.settingslib.display;
 
-import android.os.SystemProperties;
 import android.util.MathUtils;
 
-import com.android.internal.display.BrightnessSynchronizer;
-
 public class BrightnessUtils {
 
-    private static final boolean sysUseLowGamma = SystemProperties.getBoolean(
-            "persist.sys.brightness.low.gamma", false);
-
     public static final int GAMMA_SPACE_MIN = 0;
-    public static final int GAMMA_SPACE_MAX = sysUseLowGamma ? 255 : 65535;
+    public static final int GAMMA_SPACE_MAX = 65535;
 
     // Hybrid Log Gamma constant values
     private static final float R = 0.5f;
@@ -81,10 +75,6 @@ public class BrightnessUtils {
      * @return The corresponding setting value.
      */
     public static final float convertGammaToLinearFloat(int val, float min, float max) {
-        if (sysUseLowGamma) {
-            return MathUtils.constrain(BrightnessSynchronizer.brightnessIntToFloat(val), min, max);
-        }
-
         final float normalizedVal = MathUtils.norm(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, val);
         final float ret;
         if (normalizedVal <= R) {
@@ -137,10 +127,6 @@ public class BrightnessUtils {
      * @return The corresponding slider value
      */
     public static final int convertLinearToGammaFloat(float val, float min, float max) {
-        if (sysUseLowGamma) {
-            return BrightnessSynchronizer.brightnessFloatToInt(MathUtils.constrain(val, min, max));
-        }
-
         // For some reason, HLG normalizes to the range [0, 12] rather than [0, 1]
         final float normalizedVal = MathUtils.norm(min, max, val) * 12;
         final float ret;
-- 
2.34.1

